src/object_accessor.hpp

```diff
@@ -246,23 +250,38 @@ Object Object::create(ContextType ctx, SharedRealm realm, const ObjectSchema &ob
         row_index = get_for_primary_key_impl(ctx, *table, *primary_prop, primary_value);
 
         if (row_index == realm::not_found) {
-            row_index = table->add_empty_row();
             created = true;
-            if (primary_prop->type == PropertyType::Int)
+            if (primary_prop->type == PropertyType::Int) {
+#if REALM_ENABLE_SYNC
+                row_index = sync::create_object_with_primary_key(realm->read_group(), *table, Accessor::to_long(ctx, primary_value));
```

Comment:
@simonask So if user is using the sync version software, but he also need a non-sync Realm storage at the same time. What will happen? The non-sync version realm file he created cannot be read by the core without sync?